GoogleSignIn/Sources/GIDVerifyAccountDetail/Implementations/GIDVerifiedAccountDetailResult.m

```diff
@@ -16,5 +16,92 @@
 
 #import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiedAccountDetailResult.h"
 
+#import "GoogleSignIn/Sources/Public/GoogleSignIn/GIDVerifiableAccountDetail.h"
+
+#ifdef SWIFT_PACKAGE
+@import AppAuth;
+#else
+#import <AppAuth/OIDAuthState.h>
+#import <AppAuth/OIDAuthorizationRequest.h>
+#import <AppAuth/OIDAuthorizationResponse.h>
+#import <AppAuth/OIDAuthorizationService.h>
+#import <AppAuth/OIDScopeUtilities.h>
+#import <AppAuth/OIDTokenRequest.h>
+#import <AppAuth/OIDTokenResponse.h>
+#endif
+
+#if TARGET_OS_IOS && !TARGET_OS_MACCATALYST
+
+NS_ASSUME_NONNULL_BEGIN
+
 @implementation GIDVerifiedAccountDetailResult
+
+- (instancetype)initWithLastTokenResponse:(OIDTokenResponse *)tokenResponse
+                           accountDetails:(NSArray<GIDVerifiableAccountDetail *> *)accountDetails
+                                authState:(OIDAuthState *)authState {
+  self = [super init];
+  if (self) {
+    _expirationDate = tokenResponse.accessTokenExpirationDate;
+    _accessTokenString = tokenResponse.accessToken;
+    _refreshTokenString = tokenResponse.refreshToken;
+    _verifiedAccountDetails = accountDetails;
+    _verifiedAuthState = authState;
+  }
+  return self;
+}
+
+- (void)refreshTokensWithCompletion:(nullable void (^)(GIDVerifiedAccountDetailResult *,
+                                                      NSError *))completion {
+  OIDAuthorizationResponse *authResponse = self.verifiedAuthState.lastAuthorizationResponse;
+  OIDAuthorizationRequest *request = authResponse.request;
+
+  OIDTokenRequest *refreshRequest = 
+      [[OIDTokenRequest alloc] initWithConfiguration:request.configuration
+                                           grantType:OIDGrantTypeAuthorizationCode
+                                   authorizationCode:authResponse.authorizationCode
+                                         redirectURL:request.redirectURL
+                                            clientID:request.clientID
+                                        clientSecret:request.clientSecret
+                                               scope:request.scope                     
+                                        refreshToken:self.refreshTokenString
+                                        codeVerifier:request.codeVerifier
+                                additionalParameters:request.additionalParameters];
+
+  [OIDAuthorizationService performTokenRequest:refreshRequest
+                 originalAuthorizationResponse:authResponse
+                                      callback:^(OIDTokenResponse * _Nullable tokenResponse, 
+                                                 NSError * _Nullable error) {
+    if (tokenResponse) {
+      [self.verifiedAuthState updateWithTokenResponse:tokenResponse error:nil];
```

Comment:
I don't think all of the properties of `GIDVerifiedAccountDetailResult` will be updated here (i.e., I don't see the access and refresh tokens updated, and don't see the expiration date getting updated either. Are these updated somewhere else, or am I missing it here?

Reply:
You're right. I've added them in when updating the verified details array so that the result object will be updated together.